skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,297 @@
+        raise ValueError("Array points less than 3 is undefined.")
+
+    return array
+
+def tps_warp(
+    image,
+    tform,
+    output_region=None,
+    interpolation_order=1,
+    grid_scaling=None
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    src to the dst, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    tform : transformation object.
+        Inverse coordinate map, which transforms coordinates in the output
+        images into their corresponding coordinates in the input image.
+    output_region : tuple of integers, optional
+        The region ``(xmin, ymin, xmax, ymax)`` of the output
+        image that should be produced. (Note: The region is inclusive, i.e.
+        xmin <= x <= xmax)
+    interpolation_order : int, optional
+        If value is 1, then use linear interpolation else use
+        nearest-neighbor interpolation.
+    grid_scaling : int, optional
+        If grid_scaling is greater than 1, say x, then the transform is
+        defined on a grid x times smaller than the output image region.
+        Then the transform is bilinearly interpolated to the larger region.
+        This is fairly accurate for values up to 10 or so.
+
+    Returns
+    -------
+    warped : array_like
+        The warped input image.
+
+    Examples
+    --------
+    Produce a warped image rotated by 90 degrees counter-clockwise:
+
+    >>> import skimage as ski
+    >>> astronaut = ski.data.astronaut()
+    >>> image = ski.color.rgb2gray(astronaut)
+    >>> src = np.array([[0, 0], [0, 500], [500, 500],[500, 0]])
+    >>> dst = np.array([[500, 0], [0, 0], [0, 500],[500, 500]])
+    >>> output_region = (0, 0, image.shape[0], image.shape[1])
+    >>> tform = ski.transform.TPSTransform()
+    >>> tform.estimate(src, dst)
+    True
+    >>> warped_image = ski.transform.tps_warp(
+    ...     image, tform, output_region=output_region
+    ... )
+
+    References
+    ----------
+    .. [1] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+    decomposition of deformations." IEEE Transactions on pattern analysis and
+    machine intelligence 11.6 (1989): 567–585.
+
+    """
+    image = np.asarray(image)
+
+    assert hasattr(tform, "transform")
+
+    if image.size == 0:
+        raise ValueError("Cannot warp empty image with dimensions", image.shape)
+    # if image.ndim != 2:
+    #     raise ValueError("Only 2-D images (grayscale or color) are supported")
+    if output_region is None:
+        output_region = (0, 0, image.shape[0], image.shape[1])
+
+    x_min, y_min, x_max, y_max = output_region
+    if grid_scaling is None:
+        grid_scaling = 1
+    x_steps = (x_max - x_min) // grid_scaling
+    y_steps = (y_max - y_min) // grid_scaling
+    x, y = np.mgrid[x_min:x_max:x_steps*1j, y_min:y_max:y_steps*1j]
+
+    transform = tform.transform(x, y)
```

Comment:
Ah, that's why you have the extra `transform` method. Can we replace this with `tform.estimate` and `tform.__call__` somehow to avoid duplication?

Reply:
I have implemented an update that addresses this concern. The changes include removing the `transform` method and updating the implemention of the `__call__` method.